pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go

```diff
@@ -86,8 +89,12 @@ func createLogicalReplicationStreamPlanHook(
 		); err != nil {
 			return err
 		}
-
 		if !stmt.Options.IsDefault() {
+			// TODO: update handling of the the returned options
+			_, err := evalLogicalReplicationOptions(ctx, stmt.Options, exprEval, &p.ExtendedEvalContext().Context, p.SemaCtx())
```

Comment:
This is durinv exec; don’t we also need to do something during typecheck ?

Reply:
I think we would really only need 
```
	toTypeCheck := []exprutil.ToTypeCheck{
		exprutil.Strings{stmt.PGURL},
		exprutil.Strings{
			stmt.Options.Cursor,
			stmt.Options.DefaultFunction,
			stmt.Options.Mode,
		},
	}
```
WDYT? A similar eval check happens inside evalLogicalReplicationOptions when eval.String is called too.